components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/dao/impl/PolicyDAOImpl.java

```diff
@@ -729,28 +723,198 @@ private static boolean isDefaultPoliciesExist(Connection connection) throws SQLE
+     * Adding api policy to db
+     *
+     * @param connection
+     * @param name
+     * @param displayName
+     * @param description
+     * @param quotaType
+     * @param quota
+     * @param unitTime
+     * @param timeUnit
+     * @param pipelines
+     * @param applicableLevel
+     * @throws SQLException
+     */
     private static void addAPIPolicy(Connection connection, String name, String displayName, String description,
-                              String quotaType, int quota, long unitTime, String timeUnit, String applicableLevel)
-                                                                                                throws SQLException {
-        final String query = "INSERT INTO AM_API_POLICY (UUID, NAME, DISPLAY_NAME, DESCRIPTION, " +
-                "DEFAULT_QUOTA_TYPE, DEFAULT_QUOTA, DEFAULT_UNIT_TIME, DEFAULT_TIME_UNIT, APPLICABLE_LEVEL) " +
-                "VALUES (?,?,?,?,?,?,?,?,?)";
+            String quotaType, int quota, long unitTime, String timeUnit, List<Pipeline> pipelines,
+            String applicableLevel) throws SQLException {
 
+        final String query = "INSERT INTO AM_API_POLICY (UUID, NAME, DISPLAY_NAME, DESCRIPTION, "
+                + "DEFAULT_QUOTA_TYPE, DEFAULT_QUOTA, DEFAULT_QUOTA_UNIT, DEFAULT_UNIT_TIME,"
+                + " DEFAULT_TIME_UNIT, APPLICABLE_LEVEL) " + "VALUES (?,?,?,?,?,?,?,?,?,?)";
+        String policyID = UUID.randomUUID().toString();
         try (PreparedStatement statement = connection.prepareStatement(query)) {
-            statement.setString(1, UUID.randomUUID().toString());
+            statement.setString(1, policyID);
             statement.setString(2, name);
             statement.setString(3, displayName);
             statement.setString(4, description);
             statement.setString(5, quotaType);
             statement.setInt(6, quota);
-            statement.setLong(7, unitTime);
-            statement.setString(8, timeUnit);
-            statement.setString(9, applicableLevel);
+            statement.setString(7, "MB");
+            statement.setLong(8, unitTime);
+            statement.setString(9, timeUnit);
+            statement.setString(10, applicableLevel);
 
             statement.execute();
+
+            if (pipelines != null) {
+                addAPIPipeline(connection, pipelines, policyID);
+            }
         }
     }
 
+    /**
+     * Adding pipelines for API policy
+     *
+     * @param connection
+     * @param pipelines
+     * @param uuid
+     * @throws SQLException
+     */
+
+    private static void addAPIPipeline(Connection connection, List<Pipeline> pipelines, String uuid)
+            throws SQLException {
+
+        final String query =
+                "INSERT INTO AM_CONDITION_GROUP (UUID, " + "QUOTA_TYPE, QUOTA, QUOTA_UNIT, UNIT_TIME, TIME_UNIT) "
+                        + "VALUES (?,?,?,?,?,?)";
+        final String conIDquery = "SELECT CONDITION_GROUP_ID FROM AM_CONDITION_GROUP WHERE UUID = ?";
+
```

Comment:
I noticed the query "SELECT CONDITION_GROUP_ID FROM AM_CONDITION_GROUP WHERE UUID" is repeated twice. Shall we write a single method to get the condition group id?